src/aws/auth.rs

```diff
@@ -165,6 +168,28 @@ fn default_profile() -> String {
 }
 
 impl AwsAuthentication {
+    pub async fn credentials_cache(&self) -> crate::Result<CredentialsCache> {
+        match self {
+            AwsAuthentication::Role {
+                load_timeout_secs, ..
+            }
+            | AwsAuthentication::Default {
+                load_timeout_secs, ..
+            } => {
+                let credentials_cache = CredentialsCache::lazy_builder()
+                    .load_timeout(
+                        load_timeout_secs
+                            .map(Duration::from_secs)
+                            .unwrap_or(DEFAULT_LOAD_TIMEOUT),
+                    )
+                    .into_credentials_cache();
+
+                Ok(credentials_cache)
+            }
+            _ => Ok(CredentialsCache::lazy()),
+        }
+    }
+
```

Comment:
Was there any manual testing done? I'm just thinking about recent AWS component regressions we had.

Reply:
Manually tested against cloudwatch logs! With and without valid credentials. Neither encountered the signing issue we previously encountered.